client/src/pages/CreateOrganizationProfile.js

```diff
@@ -0,0 +1,325 @@
+import { WhiteSpace } from "antd-mobile";
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+// import axios from "axios";
+import filterOptions from "../assets/data/filterOptions";
+import createOrganizationProfile from "../assets//data/createOrganizationProfile";
+import styled from "styled-components";
+import CustomModal from "../components/CreatePost/CustomModal";
+import RadioGroup from "../components/CreatePost/RadioGroup";
+import Heading from "components/Typography/Heading";
+import DownArrowButton from "../components/Button/DownArrowButton";
+import Input from "../components/Input/BaseInput";
+import SubmitButton from "../components/Button/SubmitButton";
+import Label from "../components/Input/Label";
+import StyledCheckbox from "../components/Input/Checkbox";
+import { theme } from "../constants/theme";
+import {
+  CreatePostWrapper,
+  StyledForm,
+} from "../components/CreatePost/StyledCreatePost";
+
+const InputWrapper = styled.div`
+  width: 100%;
+`;
+
+const StyleLabel = {
+  textAlign: "left",
+};
+
+const StyleInput = {
+  width: "100%",
+};
+
+const types = Object.values(filterOptions)[2].options;
+const { type, industry } = createOrganizationProfile;
+
+const initialState = {
+  state: {
+    modal: false,
+    options: [],
+    selected: "",
+  },
+  formData: {
+    title: "",
+    description: "",
+    tags: [],
+    type: type.default.value,
+    industry: industry.default.value,
+  },
+  errors: [],
+};
+
+const errorMsg = {
+  name: "Please include an organization name.",
+  email: "Please include an organization email.",
+  location: "Please include a location.",
+  type: "Please include an organization type",
+  industry: "Please include an organization industry",
+  interest: "Please include what your looking for",
+  privacy: "Please include our Privacy Policy",
+  conditions: "Please include Terms and Conditions",
+};
+
+const CreateOrgProfile = (props) => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [location, setLocation] = useState("");
+  const [global, setGlobal] = useState("");
+  const [volunteer, setVolunteer] = useState("");
+  const [staff, setStaff] = useState("");
+  const [investors, setInvestors] = useState("");
+  const [donations, setDonations] = useState("");
+  const [other, setOther] = useState("");
+  const [privacy, setPrivacy] = useState("");
+  const [conditions, setConditions] = useState("");
+  const [state, setState] = useState(initialState.state);
+  const [formData, setFormData] = useState(initialState.formData);
+  const [errors, setErrors] = useState(initialState.errors);
+  const { modal, selected, options } = state;
+
+  const handleInputChangeName = (e) => {
+    setName(e.target.value);
```

Comment:
I think we should use the `react-hook-form` library to optimize the forms, I was also adding this to login & signup form because I saw it was used in some other forms and it prevents us from reinventing the wheel and having to do all this stuff manually

Reply:
Right, forgot about `react-hook-forms`. Good idea, maybe we should let everyone know to use it for any forms.

Reply:
With react-hook-forms the page only renders once when handleSubmit is used from react-hook-forms. This is a lot more efficient and less costly compare to making a re-render every time a value changes